superset/databases/api.py

```diff
@@ -920,13 +939,25 @@ def available(self) -> Response:
 
             available_databases.append(payload)
 
-        available_databases.sort(
-            key=lambda payload: preferred_databases.index(payload["engine"])
-            if payload["engine"] in preferred_databases
-            else len(preferred_databases)
```

Comment:
This was sorting the preferred DBs correctly, but doing a non-deterministic sort on the non-preferred. 🤦